File: Niuke_Introduction/AB1/AB1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

type Stack struct {
	items []string
}

func (s *Stack) IsEmpty() bool {
	return len(s.items) < 1
}
func (s *Stack) Size() int {
	return len(s.items)
}
func (s *Stack) Pop() string {
	if !s.IsEmpty() {
		a := s.items[s.Size()-1]
		s.items = s.items[:s.Size()-1]
		return a
	} else {
		return "error"
	}
}
func (s *Stack) Push(a string) {
	s.items = append(s.items, a)
}
func (s *Stack) Top() string {
	if !s.IsEmpty() {
		return s.items[s.Size()-1]
	} else {
		return "error"
	}
}
func main() {
	var s Stack
	var reader = bufio.NewReader(os.Stdin)
	num, _ := reader.ReadString('\n')
	numInt, _ := strconv.Atoi(num[0:1])
	// fmt.Println(len(num))
	// for i, c := range num {
	// 	fmt.Printf("%d:%d  ", i, int(c))
	// }

	// fmt.Scanf(&num)
	// fmt.Printf(num)
	for i := 0; i < numInt; i++ {
		var a string
		a, _ = reader.ReadString('\n')

		if len(a) > 3 {
			if a[0:3] == "pop" {
				fmt.Println(s.Pop())
			} else if a[0:3] == "top" {
				fmt.Println(s.Top())
			}
		}
		if len(a) > 4 {
			if a[0:4] == "push" {
				// fmt.Println(a[len(a)-3 : len(a)-2])
				s.Push(a[len(a)-3 : len(a)-2])
			}
		}
	}
}

File: Niuke_AB/AB1/AB1.py
class Stack():
    def __init__(self):
        self.items=[]
    def isEmpty(self):
        return self.items==[]
    def pop(self):
        if self.size()>0:
            return self.items.pop()
        else:
            return "error"
    def top(self):
        if self.size()>0:
            return self.items[-1]
        else:
            return "error"
    def push(self,item):
        self.items.append(item)
    def size(self):
        return len(self.items)
s=Stack()
num=int(input())
for i in range(num):
    a=input()
    a=a.split(' ')
    b=None
    if a[0]=="push":
        s.push(a[1])
    elif a[0]=="top":
        print(s.top())
    elif a[0]=="pop":
        print(s.pop())